### PR TITLE
Update dashboard

### LINKED ISSUE #1732: Bazel Dashboard - several issues with rules_repos
- missing rules_cc, rules_android, rules_python
- possibly also add rules_rust
- sorting (I’d wish rules_repos are together, right now rules_pkg is before bazel and rules_java after), 
- a lot of rules_* don’t have categories:  "untriaged", (or at least "no priority"), "need more information", "need review"
- owner changes: 
  - rules_java->hvd, 
  - rules_cc->gnish, 
  - stardoc->arostovtsev
  - rules_python -> rlevasseur

### PR DESCRIPTION
**Pull request by coeuvre**
Check in bunch of fixes/updates:

- Sort repo by names when displaying them on the sidebar.
- Use JDK20.
- Remove duplicated `team_owner` field in `github_team_table`. Make it easier to change owners of repo.
- Use case-insensitive text for labels to match the search result on Github.
- Fix a bug that sometimes action owners for an issue are not correctly computed.
- Update the logic for determining whether an issue/PR is triaged according to today's triaging process.
- Update nagging emails to have a separated PRs section.
- Improve sync process.

Fixes #1732.

### CODE REVIEW

File: dashboard/server/src/main/java/build/bazel/dashboard/github/issuestatus/GithubIssueStatusService.java [hunk 1]
```diff
@@ -124,70 +125,64 @@ static Status newStatus(GithubRepo repo, GithubIssue.Data data) {
     Instant updatedAt = data.getUpdatedAt();
 
     switch (status) {
-      case TO_BE_REVIEWED:
-      case MORE_DATA_NEEDED:
-      case REVIEWED:
+      case TO_BE_REVIEWED, MORE_DATA_NEEDED, REVIEWED -> {
```
Comment (meisterT) [inline dashboard/server/src/main/java/build/bazel/dashboard/github/issuestatus/GithubIssueStatusService.java:142]: I think switch statements allow you to be even more concise now, e.g. write here:

```
case TO_BE_REVIEWED, MORE_DATA_NEEDED, REVIEWED -> updatedAt.plus(7, DAYS);
```

Similar for the `default` case below

File: dashboard/server/src/main/java/build/bazel/dashboard/github/issuestatus/GithubIssueStatusService.java [hunk 2]
```diff
@@ -124,70 +125,64 @@ static Status newStatus(GithubRepo repo, GithubIssue.Data data) {
...
-    return null;
   }
 
   private ImmutableList<String> findActionOwner(
       GithubRepo repo, GithubIssue issue, GithubIssue.Data data, Status status) {
     switch (status) {
-      case TO_BE_REVIEWED:
-        return ImmutableList.of();
-      case MORE_DATA_NEEDED:
+      case MORE_DATA_NEEDED -> {
```
Comment (meisterT) [inline dashboard/server/src/main/java/build/bazel/dashboard/github/issuestatus/GithubIssueStatusService.java:172]: same for this switch

**Review by meisterT — APPROVED** [review]

File: dashboard/server/src/main/java/build/bazel/dashboard/github/issuestatus/GithubIssueStatusService.java (hunk 1 above)
Comment (coeuvre) [reply dashboard/server/src/main/java/build/bazel/dashboard/github/issuestatus/GithubIssueStatusService.java]: Done.

File: dashboard/server/src/main/java/build/bazel/dashboard/github/issuestatus/GithubIssueStatusService.java (hunk 2 above)
Comment (coeuvre) [reply dashboard/server/src/main/java/build/bazel/dashboard/github/issuestatus/GithubIssueStatusService.java]: Done.

**Pull request merged**